### ISSUE TITLE
Incompatibility with ReplayMod

### ISSUE THREAD

**Issue by Johni0702** [Pokechu22/WorldDownloader#112]
When opening a replay in the [ReplayMod] with the 4.0.3.3-mc1.12.2 LiteLoader version of WDL installed (probably all other MC versions as well), the game crashes with:
```
java.lang.RuntimeException: WorldDownloader: Couldn't find any Field of type "interface io.netty.channel.Channel" from class "class com.replaymod.replay.ReplayHandler$1" !
	at wdl.ReflectionUtils.findField(ReflectionUtils.java:91) ~[ReflectionUtils.class:?]
	at wdl.ReflectionUtils.findAndGetPrivateField(ReflectionUtils.java:118) ~[ReflectionUtils.class:?]
	at net.minecraft.client.network.NetHandlerPlayClient.handler$init$zzd000(NetHandlerPlayClient.java:2266) ~[brz.class:?]
	at net.minecraft.client.network.NetHandlerPlayClient.<init>(NetHandlerPlayClient.java:287) ~[brz.class:?]
	at com.replaymod.replay.ReplayHandler.setup(ReplayHandler.java:179) ~[ReplayHandler.class:?]
	at com.replaymod.replay.ReplayHandler.<init>(ReplayHandler.java:105) ~[ReplayHandler.class:?]
```
https://github.com/Pokechu22/WorldDownloader/blob/9d2d900e23ded053cc631c10fadec10b356462ff/1.12.2-litemod/src/main/java/com/uyjulian/LiteModWDL/mixin/MixinNetHandlerPlayClient.java#L49

which is to be expected considering the ReplayMod uses an extended NetworkManager class during replay.
If I'm not mistaken, the fix for this should be as simple as replacing the call to `findAndGetPrivateField(networkManagerIn, Channel.class)` with `findAndGetPrivateField(networkManagerIn, NetworkManager.class, Channel.class)`

[ReplayMod]: https://github.com/ReplayMod/ReplayMod/tree/stable

**Comment by Pokechu22** [comment]
Is this crash specifically only when playing back replays, or does it also happen during normal gameplay/saving replays?  I haven't messed with replaymod much though I do understand the basic idea of how it works.

The fix you've included looks like it would be correct in terms of fixing the crash, but I'm not 100% sure I even want to be running that code if it only applies during playback (though, also, I don't know how I'd sanely detect that case; checking if the network manager class doesn't match exactly seem correct and I don't think there's any harm to running the code).

**Comment by Pokechu22** [comment]
I've applied the fix you wrote.  I haven't tested how it works with replaymod yet (though I did confirm that it doesn't break code in normal cases via my unit tests).  Thanks for the report!

**Comment by NPCtom** [comment]
Also why does this have the "litemod-only" tag. Doesn't WDL REQUIRE LiteLoader and not Forge.

**Comment by Pokechu22** [comment]
It's as opposed to the base edit version (which is incompatible with everything, so this issue wouldn't apply to it).  Probably not a super useful label, but I set the labels up a long time ago and now I'm just using them :P

**Comment by Pokechu22** [comment]
This should be fixed as of [v4.0.4.0](https://github.com/Pokechu22/WorldDownloader/releases/tag/v4.0.4.0).  Let me know if it's still an issue.
